service/sign/handlers_sign.go

```diff
@@ -133,6 +177,54 @@ func Serial(w http.ResponseWriter, r *http.Request) response.ErrorResponse {
 		// to the brand public key(s) for models
 	}
 
+	if isRemodelingSerialRequest(assertion) {
+		originalBrandID := assertion.HeaderString("original-brand-id")
+		originalModel := assertion.HeaderString("original-model")
+		originalSerial := assertion.HeaderString("original-serial")
+
+		if modelAssert == nil {
+			const msg = "Model assertion can't be empty for a remodeling request"
+			svlog.Message("SIGN", "invalid-assertion", msg)
+			return response.ErrorResponse{Success: false, Code: response.ErrorInvalidAssertion.Code, Message: msg, StatusCode: http.StatusBadRequest}
+		}
+
+		// Double check the serial assertion
+		serialAssert, ok := assertions["serial"]
+		if !ok {
+			const msg = "The current serial assertion can't be empty"
+			svlog.Message("SIGN", "invalid-assertion", msg)
+			return response.ErrorResponse{Success: false, Code: response.ErrorInvalidAssertion.Code, Message: msg, StatusCode: http.StatusBadRequest}
+		}
+
+		// Validate the original model by checking that it exists on the database
+		originalModelAssert, errResponse := findModel(originalBrandID, originalModel, originalSerial, apiKey)
+		if !errResponse.Success {
+			svlog.Message("SIGN", "invalid-assertion", "original model is not valid")
+			return errResponse
+		}
+
+		// Validate the new model: it must be defind in the sub-store of the orignal model
+		substore, err := datastore.Environ.DB.GetSubstore(originalModelAssert.ID, originalSerial)
+		if err != nil {
+			svlog.Message("PIVOT", "invalid-substore", "Cannot find sub-store mapping for the model")
+			return response.ErrorInvalidSubstore
+		}
+
+		// Check if find model maches requested model
+		if modelAssert.HeaderString("model") != substore.ModelName {
+			const msg = "Requested model is invalid"
+			svlog.Message("SIGN", "invalid-assertion", msg)
+			return response.ErrorResponse{Success: false, Code: response.ErrorInvalidAssertion.Code, Message: msg, StatusCode: http.StatusBadRequest}
+		}
+
+		// Don't know what to check for the original serial
+		if serialAssert.HeaderString("model") != originalModel {
```

Comment:
serial-request is self signed, I think/hope we have code to check that?  the model assertion we get is mostly for information ATM.

But for the old serial we need need to check it quite thoroughly, that it matches  all the original-* fields,   that the device key is the same between serial-request and old serial.  And also we need to actually check the signature of old serial,  it came from this vault under current assumptions so we should have the key that signed it around to check it.